ttnn/cpp/ttnn/operations/matmul/device/matmul_op.cpp

```diff
@@ -1120,7 +1120,10 @@ void Matmul::validate(
                         TT_FATAL(program_config.out_subblock_w == per_core_N || program_config.out_subblock_h == 1, "Error");
                     }
                 }
-                TT_FATAL(input_tensor_b.memory_config().memory_layout == TensorMemoryLayout::INTERLEAVED, "Error");
+                if (input_tensor_b.memory_config().is_sharded()) {
+                    TT_FATAL(this->output_mem_config.memory_layout == TensorMemoryLayout::WIDTH_SHARDED, "Operand B can only be interleaved or width sharded.");
```

Comment:
These checks will allow DRAM width sharded as well.

Please add tests for that scenario as well.

Reply:
@johanna-rock-tt  this won't work for DRAM sharded right? question here is do we add support for it or skip DRAM sharding

Reply:
Yeah, this won't work for DRAM sharded. I don't think it makes a lot of sense in the combination with DRAM sharded matmuls, right? I believe we wouldn't have any use case where we want to use the DRAM sharded Matmul (12 cores) and read the weights from L1 in stead of DRAM. I'll fix the validate to allow it only for matmul 1D.

Reply:
In1 sharding should now only be valid for matmul with in0 mcast and not DRAM sharded matmuls, could you please check weather you think that's the correct location for the checks now?

Reply:
validate() is the correct location for all of these checks.